daq/topology.py

```diff
@@ -439,17 +439,21 @@ def _add_dot1x_incoming_rule(self, incoming_acl, secondary_acl):
                     vlan = self._get_port_vlan(device.port.port_no)
                     test_ports = device.gateway.get_possible_test_ports()
                     if test_ports:
-                        self._add_dot1x_allow_rule(incoming_acl, test_ports, vlan_vid=vlan)
+                        self._add_dot1x_allow_rule(incoming_acl,
+                                                   test_ports, vlan=vlan, action='vlan_vid')
                     device_port = device.port.port_no
                     if device_port:
-                        self._add_dot1x_allow_rule(secondary_acl, [device_port], vlan_vid=vlan)
+                        self._add_dot1x_allow_rule(secondary_acl,
+                                                   [device_port], vlan=vlan, action='in_vlan')
 
-    def _add_dot1x_allow_rule(self, acl, ports, vlan_vid=None, out_vid=None):
+    def _add_dot1x_allow_rule(self, acl, ports, vlan=None, action=None):
```

Comment:
this is a somewhat clumsy way of doing this IMHO -- I'd much rather it keep the named arguments approach and just have vlan_vid=, out_vid=, in_vlan=  and use that for the if statements below.   Was there a particular reason you wanted to expand it out into the case-statement format?  Functionally the same, but it pushes more of the complexity to the function rather than the call site.

Reply:
Lint complains if I simply add a variable. I just did this to get past it but I guess I could add an ignore flag as well. Just a tradeoff at the end of the day. Should I be going the ignore flag route?